src/group_theory/torsion.lean

```diff
@@ -75,3 +77,42 @@ exponent_exists_iff_ne_zero.mpr $
 @[to_additive is_add_torsion_of_fintype]
 lemma is_torsion_of_fintype [fintype G] : is_torsion G :=
 exponent_exists.is_torsion $ exponent_exists_iff_ne_zero.mpr exponent_ne_zero_of_fintype
+
+section infinite_torsion_groups
+
+/-- The group ℚ ⧸ ℤ. -/
+@[reducible] def qmodz := ℚ ⧸ (algebra.linear_map ℤ ℚ).range
+
+namespace qmodz
+
+open rat
+
+/-- ℚ ⧸ ℤ is a torsion group -/
+lemma is_torsion : add_monoid.is_torsion qmodz :=
+λ q, begin
+  induction q using quotient.induction_on',
+  simp_rw [is_of_fin_add_order_iff_nsmul_eq_zero, submodule.quotient.mk'_eq_mk,
+    ←submodule.quotient.mk_smul, submodule.quotient.mk_eq_zero, linear_map.mem_range,
+    algebra.linear_map_apply, algebra_map_int_eq, ring_hom.eq_int_cast, nsmul_eq_mul, mul_comm],
+  exact ⟨q.denom, q.pos, q.num, mul_denom_eq_num.symm⟩,
+end
+
+end qmodz
+
+namespace polynomial
+
+open_locale polynomial
+
+variables {Fq : Type*} [field Fq] [fintype Fq]
+
+/-- Fq[X] is a torsion group -/
+lemma is_torsion : add_monoid.is_torsion Fq[X] :=
+λ f, begin
+  rw is_of_fin_add_order_iff_nsmul_eq_zero,
+  refine ⟨fintype.card Fq, fintype.card_pos, _⟩,
+  rw [nsmul_eq_smul_cast Fq _ f, finite_field.cast_card_eq_zero, zero_smul],
+end
+
+end polynomial
```

Comment:
This maybe belongs in `field_theory.finite` somewhere? Perhaps best to ask on Zulip.